com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java

```diff
@@ -745,15 +751,26 @@ private static void mainInternal(String[] args) throws IOException, CompileExcep
         System.exit(ret ? 0 : 1);
     }
 
+    private static void logErrorAndExit(Exception e) {
+        System.err.println(e.getMessage());
+        if (e.getCause() != null) {
+            System.err.println("Cause: " + e.getCause());
+        }
+        if (verbose) {
+            e.printStackTrace();
+        }
+        System.exit(1);
+    }
+
     public static void main(String[] args) throws IOException, CompileExceptionError, URISyntaxException, LibraryException {
         try {
             mainInternal(args);
-        } catch (LibraryException e) {
-            System.err.println(e.getMessage());
-            System.err.println("Cause: " + e.getCause());
-            e.printStackTrace();
-            System.exit(1);
+        } catch (LibraryException|CompileExceptionError e) {
+            logErrorAndExit(e);
+        } catch (Exception e) {
+            logErrorAndExit(e);
         }
+
```

Comment:
Also, we don't want it to look too messy if we get random errors.
We enable the traceback on "-v"